pkg/controllers/provisioning/provisioner.go

```diff
@@ -201,41 +205,45 @@ var ErrProvisionersNotFound = errors.New("no provisioners found")
 //nolint:gocyclo
 func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNodes []*state.StateNode, opts scheduler.SchedulerOptions) (*scheduler.Scheduler, error) {
 	// Build node templates
-	var machineTemplates []*scheduler.MachineTemplate
-	instanceTypes := map[string][]*cloudprovider.InstanceType{}
+	var nodeClaimTemplates []*scheduler.NodeClaimTemplate
+	instanceTypes := map[nodepoolutil.Key][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
 
-	provisionerList := &v1alpha5.ProvisionerList{}
-	if err := p.kubeClient.List(ctx, provisionerList); err != nil {
-		return nil, fmt.Errorf("listing provisioners, %w", err)
+	nodePoolList, err := nodepoolutil.List(ctx, p.kubeClient)
+	if err != nil {
+		return nil, err
 	}
-	provisionerList.Items = lo.Filter(provisionerList.Items, func(p v1alpha5.Provisioner, _ int) bool {
-		return p.DeletionTimestamp.IsZero()
+	nodePoolList.Items = lo.Filter(nodePoolList.Items, func(n v1beta1.NodePool, _ int) bool {
+		return n.DeletionTimestamp.IsZero()
 	})
-	if len(provisionerList.Items) == 0 {
+	if len(nodePoolList.Items) == 0 {
 		return nil, ErrProvisionersNotFound
 	}
 
-	// nodeTemplates generated from provisioners are ordered by weight
+	// nodeTemplates generated from NodePools are ordered by weight
 	// since they are stored within a slice and scheduling
 	// will always attempt to schedule on the first nodeTemplate
-	provisionerList.OrderByWeight()
-	for i := range provisionerList.Items {
-		provisioner := &provisionerList.Items[i]
+	nodePoolList.OrderByWeight()
+
+	for i := range nodePoolList.Items {
+		nodePool := &nodePoolList.Items[i]
+		if !nodePool.DeletionTimestamp.IsZero() {
```

Comment:
Don't we already filter this out on line 216?

Reply:
Rebase miss. Thanks for the catch